sklearn/cross_decomposition/tests/test_pls.py

```diff
@@ -126,6 +125,29 @@ def test_sanity_check_pls_regression():
     assert_array_almost_equal(y_loadings_sign_flip, y_weights_sign_flip)
 
 
+def test_sanity_check_pls_regression_dayal_macgregor():
+    # Sanity check for PLSRegression with Dayal-MacGregor
+    # The results were checked against the results of the NIPALS algorithm
+    # Keep in mind Dayal-MacGregor does not compute Y weights, scores or rotations
+
+    d = load_linnerud()
+    X = d.data
+    Y = d.target
+
+    dayal = PLSRegression(n_components=X.shape[1], algorithm="nipals")
+    nipals = PLSRegression(n_components=X.shape[1], algorithm="dayalmacgregor")
+    dayal.fit(X=X, Y=Y)
+    nipals.fit(X=X, Y=Y)
+
+    assert_array_almost_equal(np.abs(nipals.x_loadings_), np.abs(dayal.x_loadings_), 5)
+    assert_array_almost_equal(np.abs(nipals.x_weights_), np.abs(dayal.x_weights_), 5)
+    assert_array_almost_equal(
+        np.abs(nipals.x_rotations_), np.abs(dayal.x_rotations_), 5
+    )
+    assert_array_almost_equal(np.abs(nipals.y_loadings_), np.abs(dayal.y_loadings_), 5)
+    assert_array_almost_equal(nipals.predict(X), dayal.predict(X), 5)
```

Comment:
Please use `sklearn.testing.assert_allclose` in the new test instead. You can either set  `atol` if you expect values that are very close to (or exactly) zero. Otherwise, using `rtol` is most likely more meaningful.

Please do not updated the existing test that still use `assert_array_almost_equal` to keep this PR focused on the new solver to make it easier to review.

Reply:
good spot, I changed these over to use `assert_allclose` with relative tolerances instead